tests/serverpod_test_server/test_e2e/websocket_auth_test.dart

```diff
@@ -0,0 +1,65 @@
+import 'dart:async';
+
+import 'package:serverpod_test_client/serverpod_test_client.dart';
+import 'package:serverpod_test_server/test_util/config.dart';
+import 'package:serverpod_test_server/test_util/test_key_manager.dart';
+import 'package:test/test.dart';
+
+void main() {
+  group('Given a websocket that requires authentication', () {
+    TestAuthKeyManager authKeyManager = TestAuthKeyManager();
+    late Client client;
+
+    setUp(() async {
+      client = Client(
+        serverUrl,
+        authenticationKeyManager: authKeyManager,
+      );
+    });
+
+    tearDown(() async {
+      client.close();
+    });
+
+    test('Access restricted endpoint without authentication', () async {
+      await client.openStreamingConnection(
+        disconnectOnLostInternetConnection: false,
+      );
+
+      // This should be ignored by the server as user isn't authenticated.
+      await client.signInRequired.sendStreamMessage(SimpleData(num: 666));
+      var streamingStream = client.signInRequired.stream.timeout(
+        const Duration(seconds: 2),
+      );
+      var receivedDataF = streamingStream.single;
```

Comment:
Nit: is this a typo or am I missing something?
```suggestion
      var receivedData = streamingStream.single;
```